### ISSUE TITLE
Training loops redraw and save loss.png on every batch

### ISSUE THREAD

**Issue by p3zo** [p3zo/rhythmic-relationships#25]
Every training loop redraws the full loss history and writes a PNG on **every batch**:

```python
# scripts/modeling/hits_encdec/train.py:77-81  (inside the batch loop)
plt.plot(train_losses)
loss_plot_path = os.path.join(model_dir, "loss.png")
plt.tight_layout()
plt.savefig(loss_plot_path)
plt.clf()
```

`train_losses` grows by one per batch and is re-plotted in full each time, so the plotting work is quadratic in step count, on top of a synchronous PNG encode + disk write per step. On a small model this is easily the dominant cost of training.

Same pattern in: `encdec/train.py:77`, `chroma/train.py:77`, `hits_encdec_polydrums/train.py:525`, `hits_dec/train.py:241`, `hits_dec_xformer/train.py:226`, `rsp_fc/train.py:208`, `tvae/train.py:301`, `tvae/overfit_batch_tvae.py:248`.

**Fix**: move it next to the eval-interval plotting that already exists a few lines below, or gate it on `ix % eval_interval == 0`.
